dragonfly/player/player.go

```diff
@@ -1503,6 +1546,28 @@ func (p *Player) Tick(current int64) {
 		p.Hurt(4, damage.SourceVoid{})
 	}
 
+	if p.FireTicks() > 0 {
+		if p.FireProof() {
+			p.SetFireTicks(0)
+		} else {
+			p.SetFireTicks(p.FireTicks() - 1)
+		}
+		if p.FireTicks()%20 == 0 && !p.AttackImmune() {
+			p.Hurt(1, damage.SourceFireTick{})
+		}
+	}
+
+	aabb := p.AABB().Translate(p.Position())
+	for x := int(aabb.Min().X()); x <= int(aabb.Max().X()); x++ {
+		for y := int(aabb.Min().Y()); y <= int(aabb.Max().Y()); y++ {
+			for z := int(aabb.Min().Z()); z <= int(aabb.Max().Z()); z++ {
+				if collide, ok := p.World().Block(world.BlockPos{x, y, z}).(block.EntityColliding); ok {
+					collide.EntityCollide(p)
+				}
+			}
+		}
+	}
```

Comment:
Could you add a TODO to investigate moving this to `Move()` once we get proper movement interpolation?